### ISSUE TITLE
Using gallery with non-top level baseURL

### ISSUE THREAD

**Issue by tcwilkinson** [liwenyip/hugo-easy-gallery#8]
Perhaps I have not been able to find the right way to pass, but I am having difficulty getting this to work when the base url of the site is not the top-level. 

For example: my site is going to live at /foo/bar/ and config.toml is set to:
`baseurl = "http://website.com/foo/bar"`

Static images from /static/images/ therefore end up in http://website.com/foo/bar/images/* . But the resultant page expects the files to be found in http://website.com/images/* when given the shortcode: `{{< gallery dir="/images/" />}} {{< load-photoswipe >}}`. The page is create but all image <div> are blank.

If this is changed to `{{< gallery dir="/foo/bar/images/" />}} {{< load-photoswipe >}}` (even if one could pass the "/foo/bar" dynamically) then it looks for the images in /static/foo/bar/images ... which of course doesn't exist. Hence no <img>s are created on the page.

Any ideas?
Toby

PS:- this is really a great tool for hugo. Thanks for sharing it!

**Comment by tcwilkinson** [comment]
As a temporary workaround, I inserted the following into gallery.html
```
{{ $siteURL := .Site.BaseURL }}
```
[snip]
```
				{{- $linkURL := print $siteURL "/" ($.Get "dir") "/" .Name | absURL }}<!-- absolute URL to hi-res image -->
				{{- $thumbURL := print $siteURL "/" ($.Get "dir") "/" $thumb | absURL }}<!-- absolute URL to thumbnail image -->
```

It works, although not sure if it isn't a little "hacky"...

**Comment by liwenyip** [comment]
Thanks @tcwilkinson, this is an oversight on my part. 

Probably more efficient to do something like 

```
{{ $imgBaseURL := print .Site.BaseURL "/" ($.Get "dir") "/" }}
```
[snip]
```
   {{- $linkURL := print $imgBaseURL .Name | absURL }}<!-- absolute URL to hi-res image -->
   {{- $thumbURL := print $imgBaseURL "/" ($.Get "dir") "/" $thumb | absURL }}<!-- absolute URL to thumbnail image -->
```

I'm not in front of the correct computer for me to test this, so feel free to get something working and then submit a PR.

**Comment by jbrice** [comment]
@liwenyip  Tried your patch for this problem as I have a documentation site that runs in a sub-folder of server root. it looks as if the patch works for the background URLs in "gallery from folder" mode, but does not fix  "gallery from list" which suffers from the same problem.

Unfortunately I don't have time just now to triage this properly, but I'm sure you know your code well enough to spot why this is happening.

**Comment by liwenyip** [comment]
Thanks @jbrice 

I'm also kinda busy at the moment; so I'll add it to my to-do list. But I'd be very happy for someone else to fix and submit a PR in the meantime.

**Comment by jbrice** [comment]
Happy to put some time to it, and have had a go but "retired hurt" as I am a newcomer to Hugo/golang template coding and  finding your code somewhat cryptic, probably as a result of that.

I have been unable to find much in-depth documentation or "power user" examples of Hugo template coding. Are there any resources you would recommend?

**Comment by liwenyip** [comment]
@jbrice the Hugo documentation covers basics, and is improving all the time.

For more complicated questions, post/search in https://discourse.gohugo.io/

Good luck!

**Comment by jbrice** [comment]
Thank you! 

Am making progress. However from a newcomers point of view a lot of the documentation looks great, but in practice feels as if it has been written by people who are much too familiar with the topic they are writing about. And https://discourse.gohugo.io/ is full of great nuggets but needs a lot of excavation to get to many of them.

**Comment by jbrice** [comment]
@liwenyip 
The mists are clearing and I think that the fix is fairly simple. I have found that the best (only?) way to use the current version of Hugo (V0.25.1) in a sub-folder of the server is to create the site with "baseURL" set to the full path (domain plus path) to the site's root folder. Hugo then automatically prepends all asset URLs beginning with "/" with the "baseURL"string, thus essentially doing the job of the additional code you suggested a short while ago (but on a site-wide basis).  

However this URL completion is not foolproof and does not seem to process urls in CSS backgrounds in this way, including the one on line 13 of the current version of figure.html. So I have proposed a simple change that I have found to work OK on a non-server-root test site, when used along with  an appropriate "baseURL" in the config.
